tasks/shell.sh

```diff
@@ -0,0 +1,539 @@
+  critical "Platform: $platform"
+  critical "Platform Version: $platform_version"
+  critical "Machine: $machine"
+  critical "OS: $os"
+  critical ""
+  critical "Please detail your operating system type, version and any other relevant details"
+}
+
+# Get command line arguments
+
+if [ -n "$PT_version" ]; then
+  version=$PT_version
+fi
+
+#if [ -n "$PT_filename" ]; then
+#   cmdline_filename=$PT_filename
+#fi
+#
+#if [ -n "$PT_download_dir" ]; then
+#  cmdline_dl_dir==$PT_download_dir
+#fi
+#while getopts v:f:d:h opt
+#do
+#  case "$opt" in
+#    v)  version="$OPTARG";;
+#    f)  cmdline_filename="$OPTARG";;
+#    d)  cmdline_dl_dir="$OPTARG";;
+#    h) echo >&2 \
+#      "install_puppet_agent.sh - A shell script to install Puppet Agent > 5.0.0, assuming no dependencies
+#      usage:
+#      -v   version         version to install, defaults to \$latest_version
+#      -f   filename        filename for downloaded file, defaults to original name
+#      -d   download_dir    filename for downloaded file, defaults to /tmp/(random-number)"
+#      exit 0;;
+#    \?)   # unknown flag
+#      echo >&2 \
+#      "unknown option
+#      usage: $0 [-v version] [-f filename | -d download_dir]"
+#      exit 1;;
+#  esac
+#done
+shift `expr $OPTIND - 1`
+
+machine=`uname -m`
+os=`uname -s`
+
+# CODEREVIEW: What are we doing with this?
+if [ -f /opt/puppetlabs/puppet/VERSION ]; then
+  installed_version=`cat /opt/puppetlabs/puppet/VERSION`
+else
+  installed_version=uninstalled
+fi
+
+# Retrieve Platform and Platform Version
+if test -f "/etc/lsb-release" && grep -q DISTRIB_ID /etc/lsb-release; then
+  platform=`grep DISTRIB_ID /etc/lsb-release | cut -d "=" -f 2 | tr '[A-Z]' '[a-z]'`
+  platform_version=`grep DISTRIB_RELEASE /etc/lsb-release | cut -d "=" -f 2`
+elif test -f "/etc/debian_version"; then
+  platform="debian"
+  platform_version=`cat /etc/debian_version`
+elif test -f "/etc/redhat-release"; then
+  platform=`sed 's/^\(.\+\) release.*/\1/' /etc/redhat-release | tr '[A-Z]' '[a-z]'`
+  platform_version=`sed 's/^.\+ release \([.0-9]\+\).*/\1/' /etc/redhat-release`
+
+  #If /etc/redhat-release exists, we act like RHEL by default. Except for fedora
+  if test "$platform" = "fedora"; then
+    platform="fedora"
+  else
+    platform="el"
+  fi
+elif test -f "/etc/system-release"; then
+  platform=`sed 's/^\(.\+\) release.\+/\1/' /etc/system-release | tr '[A-Z]' '[a-z]'`
+  platform_version=`sed 's/^.\+ release \([.0-9]\+\).*/\1/' /etc/system-release | tr '[A-Z]' '[a-z]'`
+  # amazon is built off of fedora, so act like RHEL
+  if test "$platform" = "amazon linux ami"; then
+    platform="el"
+    platform_version="6.0"
+  fi
+# Apple OS X
+elif test -f "/usr/bin/sw_vers"; then
+  platform="mac_os_x"
+  # Matching the tab-space with sed is error-prone
+  platform_version=`sw_vers | awk '/^ProductVersion:/ { print $2 }'`
+
+  major_version=`echo $platform_version | cut -d. -f1,2`
+  case $major_version in
+    "10.6") platform_version="10.6" ;;
```

Comment:
These are ancient, I don't think we even support them anymore. But I guess macOS work is a separate ticket.